connection/cluster/ClusterClient.java

```diff
@@ -294,13 +304,17 @@ private ClusterDatabase.Replica seekPrimaryReplica() {
         }
 
         private ClusterDatabase fetchDatabaseReplicas() {
-            for (String serverAddress : clusterServerClients.keySet()) {
+            roundRobinIndex = (roundRobinIndex + 1) % clusterAddresses.size();
+            for (int i=0 ; i < clusterAddresses.size() ; i++) {
+                String serverAddress = clusterAddresses.get((roundRobinIndex + i) % clusterAddresses.size());
```

Comment:
It might be worth it to also advance the round robin index on failure here. Otherwise, if node 0 is unavailable, we effectively query node 1 twice before querying node 2.